src/allowance.ts

```diff
@@ -26,19 +26,14 @@ export class Allowance {
     this.spender = spender;
     this.provider = provider;
 
-    switch (this.token) {
-      case 'R':
-      case 'wstETH':
-        this.tokenContract = ERC20Permit__factory.connect(RaftConfig.getTokenAddress(this.token), this.provider);
-        break;
+    const tokenConfig = RaftConfig.networkConfig.tokenTickerToTokenConfigMap[token];
 
-      case 'ETH':
-        // ETH is not a contract
-        this.tokenContract = null;
-        break;
-
-      default:
-        this.tokenContract = ERC20__factory.connect(RaftConfig.getTokenAddress(this.token), this.provider);
+    if (tokenConfig.ticker === 'ETH') {
+      this.tokenContract = null;
+    } else if (tokenConfig.supportsPermit) {
+      this.tokenContract = ERC20Permit__factory.connect(RaftConfig.getTokenAddress(this.token), this.provider);
+    } else {
+      this.tokenContract = ERC20__factory.connect(RaftConfig.getTokenAddress(this.token), this.provider);
```

Comment:
Looks like it can be extracted to a token contract factory since it is also used for balance.